src/bastion/azext_bastion/custom.py

```diff
@@ -0,0 +1,327 @@
+                  "try opening the JSON view of your resource (in the Overview tab), and copying the full resource ID."
+        raise InvalidArgumentValueError(err_msg)
+
+    from .aaz.latest.network.bastion import Show
+    bastion = Show(cli_ctx=cmd.cli_ctx)(command_args={
+        "resource_group": resource_group_name,
+        "name": bastion_host_name
+    })
+
+    if bastion['sku']['name'] == "Basic" or \
+            bastion['sku']['name'] == "Standard" and bastion['enableTunneling'] is not True:
+        raise ClientRequestError('Bastion Host SKU must be Standard and Native Client must be enabled.')
+
+    if platform.system() == "Windows":
+        if disable_gateway:
+            tunnel_server = _get_tunnel(cmd, resource_group_name, bastion_host_name, target_resource_id, resource_port)
+            t = threading.Thread(target=_start_tunnel, args=(tunnel_server,))
+            t.daemon = True
+            t.start()
+            command = [_get_rdp_path(), f"/v:localhost:{tunnel_server.local_port}"]
+            launch_and_wait(command)
+            tunnel_server.cleanup()
+        else:
+            profile = Profile(cli_ctx=cmd.cli_ctx)
+            access_token = profile.get_raw_token()[0][2].get("accessToken")
+            logger.debug("Response %s", access_token)
+
+            web_address = f"https://{bastion['dnsName']}/api/rdpfile?resourceId={target_resource_id}" \
+                          f"&format=rdp&rdpport={resource_port}&enablerdsaad={enable_mfa}"
+            headers = {
+                "Authorization": f"Bearer {access_token}",
+                "Accept": "*/*",
+                "Accept-Encoding": "gzip, deflate, br",
+                "Connection": "keep-alive",
+                "Content-Type": "application/json"
+            }
+            response = requests.get(web_address, headers=headers)
+            if not response.ok:
+                raise ClientRequestError("Request to EncodingReservedUnitTypes v2 API endpoint failed.")
+
+            _write_to_file(response)
+
+            rdpfilepath = os.getcwd() + "/conn.rdp"
+            command = [_get_rdp_path()]
+            if configure:
+                command.append("/edit")
+            command.append(rdpfilepath)
+            launch_and_wait(command)
+    else:
+        raise UnrecognizedArgumentError("Platform is not supported for this command. Supported platforms: Windows")
+
+
+def _write_to_file(response):
+    with open("conn.rdp", "w", encoding="utf-8") as f:
+        for line in response.text.splitlines():
+            if not line.startswith('signscope'):
+                f.write(line + "\n")
+
+
+def _get_tunnel(cmd, resource_group_name, name, vm_id, resource_port, port=None):
+    from .tunnel import TunnelServer
+    from .aaz.latest.network.bastion import Show
+
+    bastion = Show(cli_ctx=cmd.cli_ctx)(command_args={
+        "resource_group": resource_group_name,
+        "name": name
+    })
```

Comment:
The output of bastion will apply client flatten by default. Did you verify whether it can work directly put it in TunnelServer function?

Reply:
`tunnel.py` has been modified to adapt it, e.g., https://github.com/necusjz/azure-cli-extensions/blob/898149bdb11ee5aff19bcd4fb844804ad89ee2e3/src/bastion/azext_bastion/tunnel.py#L91.